### PR TITLE
fix: iOS and Android integration SDKs compatibility updates

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
## Description of the change

- Make iOS integration SDKs compatible with both dynamic and static frameworks: Now all the RudderStack React Native integrations SDKs (core React Native SDK was already compatible) are compatible, when any of the following things are added in Podfile of the sample iOS app:
```
use_frameworks!
use_frameworks! :linkage=> :static
use_frameworks! :linkage => :dynamic
// Or, omit use_frameworks!
```
- Migrate `nx` to the latest version: This updates various packages and project.json files.
- `react-native` is updated to the recent version `0.72.4`: Which also adds the support for Gradle version 8.
- Make all Android integration SDKs compatible with Gradle version 8: This requires a simple change in all the Android integration SDKs i.e., changing the deprecated `classifier` to `archiveClassifier`.
- Updated node version in `nvmrc` file.
- Created the `Example` app from scratch: I first removed the example app and then created a new one and made the necessary changes over there. This is necessary to solve the framework issue that occurred in iOS.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: libs/rudder-integration-braze-react-native/project.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "tags": [],
   "targets": {
     "build": {
-      "executor": "@nrwl/web:rollup",
+      "executor": "@nx/rollup:rollup",
```
Comment (bardisg) [inline libs/rudder-integration-braze-react-native/project.json:9]: is that correct? should we not use https://nx.dev/nx-api/web here?

Comment (1abhishekpandey) [reply libs/rudder-integration-braze-react-native/project.json:9]: After updating the `nx` to the latest version, `@nrwl/web:rollup` is no longer working, due to that `npm run build` command got failed. So, after some research, I used `@nx/rollup:rollup` and now the build is successful. Hence, I think, it is correct.

Coming to the `@nx/web`: Although I don't have proper knowledge about it, I tried and it doesn't seem to work and the build is failing. (I replaced `@nx/rollup:rollup` with `@nx/web` and `@nx/web:rollup` and build failed).

Moreover, we need to use `rollup`, which previously came bundle inside `@nrwl/web`-`executor.json` file. Whereas, in the latest update, that is missing.

If would like me to try with something specific, please let me know.

**Review by desusai7 — APPROVED** [review]

File: libs/rudder-integration-braze-react-native/project.json (hunk 1 above)
Comment (bardisg) [reply libs/rudder-integration-braze-react-native/project.json:9]: I would help to know what the build error was. Now i have to checkout and try to build my self to figure it out

Comment (bardisg) [reply libs/rudder-integration-braze-react-native/project.json:9]: anyow seems like the latest version of web only has one executor the "file-server" so at some pont there was a breaking change. rollup is good to use for this case

**Review by bardisg — APPROVED** [review]

File: libs/rudder-integration-braze-react-native/project.json (hunk 1 above)
Comment (1abhishekpandey) [reply libs/rudder-integration-braze-react-native/project.json:9]: Sure. Let me know once you find an alternative solution for this.
Meanwhile, you can also take a look on the error, I'm getting:
image

**Pull request merged**